### ISSUE TITLE
drop color encoding in favor of default color where there is no need

### ISSUE THREAD

**Issue by Casyfill** [altair-viz/pdvega#40]
as for now, any vgplot will provide color encoding, which creates  color legend, even if there is only one category/color

**Comment by jakevdp** [comment]
I think I recall that the color encoding is only added for dataframe sources, and not for series sources. I did that because it matches the semantics of Pandas plot methods, which add a legend for dataframe.plot(), but not for series.plot().

**Comment by Casyfill** [comment]
yeah, but probably we can check and (at least by default) drop encoding if there is only one category
I can pr on that

**Comment by jakevdp** [comment]
Sure, we could do that, but that would move the pdvega behavior away from the pandas.plot behavior. I think it's less confusing to keep them similar where possible.
